### ISSUE TITLE
Matcher for creating multiple contributions

### ISSUE THREAD

**Issue by jensschuppe** [Project60/org.project60.banking#323]
Possible approaches:
- threshold (split at a given amount or percentage, enter remainder as separate contribution)
- threshold multiple (split every X units, e.g. split an amount of € 150 into three contributions of € 50 each)
- array of contributions by defining a prefix (e.g. `contr_`) for contribution properties, and setting them as e.g. `contr_1_financial_type`, `contr_2_financial_type`, etc., resulting in multiple contributions if each has enough data
- match one or multiple existing contribution(s) and enter the remainder as one of the above

**Comment by jensschuppe** [comment]
I implemented an approach slightly different from the original intention and left out reconciling with existing contributions for now. The configuration parameters are as follows:
```json
{
    "required_values": {
        "btx.financial_type_id": "type:positiveInteger"
    },
    "defaults": {
        "source": "Bank transaction processed by CiviBanking"
    },
    "contributions": [
        {
            "contribution": {
                "total_amount": 50,
                "financial_type_id": 2,
                "custom_123": "foobar"
            },
            "missing_amount_penalty": 1.0,
            "missing_parameters_penalty": 0.1
        },
        {
            "contribution": {
                "total_amount": 20,
                "financial_type_id": 1,
                "campaign_id": 123
            },
            "missing_amount_penalty": 0.1,
            "missing_parameters_penalty": 0.1
        }
    ],
    "remainder": {
        "contribution": {
            "financial_type_id": 1
        },
        "min_amount_penalty": 0.05,
        "min_amount": 1,
        "max_amount_penalty": 0.05,
        "max_amount": 100
    }
}

```
Required for all `contributions.contribution` sections is `total_amount` since this is what serves as a delimiter for splitting the transaction amount into separate contributions. All parameters will be passed to the `Contribution.create` API later on. The latter also applies to the `remainder.contribution` section.

Parameters defined within `defaults` applies for all contributions created by the matcher and will be overridden by either parameters defined within each `contributions.contribution` section or in propagated values within the transaction. The latter might cause issues, since e.g. all contributions' financial type would be overwritten by a `btx.financialy_type_id` when only one should be propagated. This should be configurable with a `propagate_value` configuration parameter inside each `contributions` and the `remainder` sections.

Suggestions will only be made if the entire transaction amount can be captured by contributions. Reasons for this to fail might be:
- missing mandatory parameters for contributions (either in `contributions` or `remainder`)
- missing `remainder` section with a remainder amount after all `contributions` sections have been processed
- missing required values as configured with the plugin

Penalties can be applied for
- missing amount for `contributions` sections
- missing parameters for `contributions` sections
- undercutting or exceeding defined minimum/maximum amounts for the `remainder`

**Comment by bjendres** [comment]
released with ``0.8.0``
